docs/pages/v2.md

```diff
@@ -0,0 +1,641 @@
+---
+title: Preparing for v2
+order: 1
+new: true
+---
+
+# Preparing for v2
+
+All v2 APIs and behaviors are available in v1 with [Future Flags][future-flags]. They can be enabled one at a time to avoid development disruption of your project. After you have enabled all flags, upgrading to v2 should be a non-breaking upgrade.
+
+## File System Route Convention
+
+#### Upgrading without changing files
+
+You can keep using the old convention with `@remix-run/v1-route-convention` even after upgrading to v2 if you don't want to make the change right now (or ever, it's just a convention and you can use whatever file organization you prefer).
+
+```shellscript
+npm i @remix-run/v1-route-convention
+```
+
+```js filename=remix.config.js
+const {
+  createRoutesFromFolders,
+} = require("@remix-run/v1-route-convention");
+
+module.exports = {
+  future: {
+    // makes the warning go away in v1.15
+    v2_routeConvention: true,
+  },
+
+  routes(defineRoutes) {
+    // uses the v1 convention, works in v1.15+ and v2
+    return createRoutesFromFolders(defineRoutes);
+  },
+};
+```
```

Comment:
@mcansh can you confirm once this is published and this API is correct?

Reply:
for basic usage, yes. we also have options for `appDirectory`, `ignoredFilePatterns`, and `routesDirectory`. 

we should also note that you'll need to opt out of having remix handling routes for you.

```js
module.exports = {
  future: {
    // makes the warning go away in v1.15
    v2_routeConvention: true,
  },

  // opt out of having remix handle your routes
  ignoredRouteFiles: ["**/*"],

  routes(defineRoutes) {
    // uses the v1 convention, works in v1.15+ and v2
    return createRoutesFromFolders(defineRoutes);
  },
};
```